### PR TITLE
Github Workflow Replacement for Jenkins Job, beam_PreCommit_Website_Stage_GCS (Cron, Commit, Phrase)

### PR DESCRIPTION
**Pull request by magicgoody**
Similar workflow migration PR as reference: https://github.com/apache/beam/pull/27042

PR migrates Jenkins Jobs to Github Actions Workflow ".github/workflows/beam_PreCommit_Website_Stage_GCS.yml":

https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Website_Stage_GCS_Commit/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Website_Stage_GCS_Cron/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Website_Stage_GCS_Phrase/

The code has been tested on Self-hosted GKE runners ubuntu 20.04.

The tests was made by creating pull requests with changes in trigger path ['website/**'], by adding a comment to the PR, by creating a push.

image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/beam_PreCommit_Website_Stage_GCS.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+    beam_PreCommit_Website_Stage_GCS:
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_Website_Stage_GCS.yml:47]: nit: formatting

Comment (magicgoody) [reply .github/workflows/beam_PreCommit_Website_Stage_GCS.yml]: fixed, thanks

**Comment by Abacn** [comment]
The main purpose of this suite is to generate a preview for the website, e.g. #27374 the run is https://ci-beam.apache.org/job/beam_PreCommit_Website_Stage_GCS_Commit/11517/ and there is a piece of description

```
PR #27374: Update Beam website to rele...
http://apache-beam-website-pull-requests.storage.googleapis.com/27374/index.html
```

There is a log in build `Website published to http://apache-beam-website-pull-requests.storage.googleapis.com/27374/index.html`

There might be still this log in the GitHub Action run if one dig into. Is it possible to show this message in a more visible way (like currently we have as the description of the jenkins run):

image

**Comment by magicgoody** [comment]
> The main purpose of this suite is to generate a preview for the website, e.g. #27374 the run is https://ci-beam.apache.org/job/beam_PreCommit_Website_Stage_GCS_Commit/11517/ and there is a piece of description
> 
> ```
> PR #27374: Update Beam website to rele...
> http://apache-beam-website-pull-requests.storage.googleapis.com/27374/index.html
> ```
> 
> There is a log in build `Website published to http://apache-beam-website-pull-requests.storage.googleapis.com/27374/index.html`
> 
> There might be still this log in the GitHub Action run if one dig into. Is it possible to show this message in a more visible way (like currently we have as the description of the jenkins run):
> 
> image

Hello @Abacn 
Thank you for you comment.
I've added the step which adds a link to the summary page. Please take a look.
image

**Review by andreydevyatkin — CHANGES_REQUESTED** [review]
see the inline comments

File: .github/workflows/beam_PreCommit_Website_Stage_GCS.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          export_default_credentials: true
+      - name: Run website_stageWebsite script
+        run: ./gradlew :website:stageWebsite --continue 
+          -Dorg.gradle.jvmargs=-Xms2g 
+          -Dorg.gradle.jvmargs=-Xmx6g 
+          -Dorg.gradle.vfs.watch=false 
+          -Pdocker-pull-licenses
+      - name: Add website link to the summary
+        run: echo "Website published to http://apache-beam-website-pull-requests.storage.googleapis.com/${{ github.event.pull_request.number }}/index.html" >> $GITHUB_STEP_SUMMARY
```
[the remark this answers is not in the document]
Comment (magicgoody) [reply .github/workflows/beam_PreCommit_Website_Stage_GCS.yml]: Fixed, link generation process is now dynamic. If it is related to a PR it has the PR number in the link, for others it is generated the same as in the provided link above. Thank you.

File: .github/workflows/beam_PreCommit_Website_Stage_GCS.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          cache-read-only: false
+      - name: Authenticate on GCP
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          service_account_email: ${{ secrets.GCP_SA_EMAIL }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          export_default_credentials: true
+      - name: Run website_stageWebsite script
+        run: ./gradlew :website:stageWebsite --continue
```
[the remark this answers is not in the document]
Comment (magicgoody) [reply .github/workflows/beam_PreCommit_Website_Stage_GCS.yml]: FIxed, thanks

**Review by andreydevyatkin — APPROVED** [review]
LGTM

**Pull request merged**